gambit/1.0.0/Dockerfile

```diff
@@ -0,0 +1,75 @@
+# Software installation, no database files
+FROM ubuntu:xenial as base
+
+ARG GAMBIT_SOFTWARE_VERSION="1.0.0"
+ARG GAMBIT_GIT_REF=v${GAMBIT_SOFTWARE_VERSION}
+ARG GAMBIT_GIT_URL=github.com/jlumpe/gambit@${GAMBIT_GIT_REF}
+ARG MINICONDA_URL=https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh
+ARG PYTHON_VERSION=3.9
+
+LABEL base.image="ubuntu:xenial"
+LABEL dockerfile.version="1"
+LABEL software="GAMBIT"
+LABEL software.version=${GAMBIT_SOFTWARE_VERSION}
+LABEL description="Rapid genomic-distance comparison for taxonomic identification of microbial pathogens"
+LABEL website="https://github.com/jlumpe/gambit"
+LABEL license="https://github.com/jlumpe/gambit/blob/master/LICENSE"
+LABEL maintainer1="Kevin Libuit"
+LABEL maintainer.email1="kevin.libuit@theiagen.com"
+LABEL maintainer2="Michelle Scribner"
+LABEL maintainer.email2="michelle.scribner@theiagen.com"
+LABEL maintainer3="Jared Lumpe"
+LABEL maintainer.email3="jared@jaredlumpe.com"
+
+# Environment
+ENV GAMBIT_DB_PATH=/gambit-db
+ENV LC_ALL=C.UTF-8
+
+# install c compiler & needed software for conda install; cleanup apt garbage
+RUN apt-get update && \
+  apt-get install -y \
+    wget \
+    gcc \
+    git \
+    build-essential \
+    && \
+  apt-get autoclean && \
+  rm -rf /var/lib/apt/lists/*
+
+# Install miniconda
+ADD $MINICONDA_URL miniconda.sh
+RUN bash ./miniconda.sh -p /miniconda -b && \
+  rm miniconda.sh
+
+# Create conda environment and add to PATH
+RUN /miniconda/bin/conda create -n gambit python=$PYTHON_VERSION numpy=1.23.1 Cython=0.29.30 pandas=1.4.3
+ENV PATH="/miniconda/envs/gambit/bin:/miniconda/bin:$PATH"
+
+# Install GAMBIT package
+RUN pip install git+https://${GAMBIT_GIT_URL} && \
+  conda clean -a -y
+
+RUN mkdir $GAMBIT_DB_PATH /data
+
+WORKDIR /data
+
+
+# With database files added
+FROM base AS app
+
+ARG GAMBIT_DB_BASE_URL=https://storage.googleapis.com/jlumpe-gambit/public/databases/refseq-curated/1.0
+ARG GAMBIT_DB_GENOMES_URL=$GAMBIT_DB_BASE_URL/gambit-refseq-curated-1.0.gdb
+ARG GAMBIT_DB_SIGNATURES_URL=$GAMBIT_DB_BASE_URL/gambit-refseq-curated-1.0.gs
+
+ADD $GAMBIT_DB_GENOMES_URL $GAMBIT_DB_PATH/
+ADD $GAMBIT_DB_SIGNATURES_URL $GAMBIT_DB_PATH/
+RUN chmod -R a+rx $GAMBIT_DB_PATH
+
+
+# Run test
+FROM app as test
+
+RUN mkdir test
+COPY test.sh test/
+RUN (cd test && bash test.sh) && \
+  rm -r test
```

Comment:
I wouldn't worry about removing test. This directory (and anything in the test layer) doesn't end up in the final downloadable product on dockerhub or quay.

Reply:
Thanks, I'm still unsure about exactly how build stages are handled. Do they just take the last one not labeled "test"?

Reply:
The one that goes on dockerhub/quay is 'app' and 'test' is required for our github actions. Anything after 'app' and before 'test' gets a little lost with our current setup.

This isn't a github or docker thing, it's a 'StaPH-B/docker-builds' thing that should probably go into the documentation somewhere.

Reply:
Cool, thanks